src/controllers/services.ts

```diff
@@ -243,3 +260,360 @@ export async function putOrganizationLogo(
+          ResponseErrorInternal(err.message)
+        )
+    )
+    .chain(
+      fromPredicate(
+        _ => _.total === 0,
+        _ =>
+          ResponseErrorConflict(
+            "A review is already in progress for the service"
+          )
+      )
+    )
+    .chain(_ =>
+      jiraClient
+        .createJiraIssue(
+          `[DISATTIVAZIONE] servizio ${serviceId}` as NonEmptyString,
+          `Effettua la disattivazione del servizio al link https://developer.io.italia.it/service/${serviceId}` as NonEmptyString,
+          serviceId,
+          ["DISATTIVAZIONE" as NonEmptyString]
+        )
+        .map(__ =>
+          ResponseSuccessJson<ReviewStatus>({
+            detail: "A new issue is created",
+            status: 200,
+            title: "Create new Issue"
+          })
+        )
+        .mapLeft(err => ResponseErrorInternal(err.message))
+    )
+    .fold<
+      | IResponseSuccessAccepted
+      | IResponseSuccessJson<ReviewStatus>
+      | IResponseErrorForbiddenNotAuthorized
+      | IResponseErrorConflict
+      | IResponseErrorInternal
+      | IResponseErrorNotFound
+    >(identity, identity)
+    .run();
+}
+
+export async function newReviewRequest(
+  apiClient: ApiManagementClient,
+  jiraClient: IJiraAPIClient,
+  authenticatedUser: AdUser,
+  serviceId: NonEmptyString,
+  jiraConfig: config.IJIRA_CONFIG
+): Promise<
+  | IResponseSuccessAccepted
+  | IResponseSuccessJson<ReviewStatus>
+  | IResponseErrorForbiddenNotAuthorized
+  | IResponseErrorConflict
+  | IResponseErrorInternal
+  | IResponseErrorNotFound
+> {
+  const maybeApimUser = await getApimUser(
+    apiClient,
+    authenticatedUser.emails[0]
+  );
+  if (isNone(maybeApimUser)) {
+    return ResponseErrorNotFound(
+      "API user not found",
+      "Cannot find a user in the API management with the provided email address"
+    );
+  }
+  const authenticatedApimUser = maybeApimUser.value;
+
+  // An admin cannot require a Service Review
+
+  if (isAdminUser(authenticatedApimUser)) {
+    return ResponseErrorForbiddenNotAuthorized;
+  }
+
+  // Authenticates this request against the logged in user
+  // checking that serviceId = subscriptionId
+  // if the user is an admin we skip the check on userId
+
+  const maybeSubscription = await getUserSubscription(
+    apiClient,
+    serviceId,
+    authenticatedApimUser.id
+  );
+  if (isNone(maybeSubscription)) {
+    return ResponseErrorNotFound(
+      "Subscription not found",
+      "Cannot get a subscription for the logged in user"
+    );
+  }
+
+  // Get old service data
+
+  /*
+  // Any idea why we need to check service fields validation?
+  const errorOrService = toEither(
+    await notificationApiClient.getService({
+      id: serviceId
+    })
+  );
+  if (isLeft(errorOrService)) {
+    return ResponseErrorNotFound(
+      "Service not found",
+      "Cannot get a service with the provided id."
+    );
+  }
+
+  const service = errorOrService.value;
+  */
```

Comment:
Old code can be removed

Reply:
Or maybe we should de-comment this code. If this API is called for an not existing service we should return 404 and doesn't create a Jira card.

Reply:
Fix with 4b23119